src/ray/core_worker/core_worker.cc

```diff
@@ -2458,29 +2456,60 @@ void CoreWorker::HandleSubscribeForObjectEviction(
   // Returns true if the object was present and the callback was added. It might have
   // already been evicted by the time we get this request, in which case we should
   // respond immediately so the raylet unpins the object.
-  if (!reference_counter_->SetDeleteCallback(object_id, respond)) {
+  if (!reference_counter_->SetDeleteCallback(object_id, unpin_object)) {
     // If the object is already evicted (callback cannot be set), unregister the
-    // subscription.
-    object_status_publisher_->UnregisterSubscription(
-        rpc::ChannelType::WORKER_OBJECT_EVICTION, subscriber_node_id, object_id.Binary());
+    // subscription & publish the message so that the subscriber knows it.
+    unpin_object(object_id);
     std::ostringstream stream;
     stream << "Reference for object " << object_id << " has already been freed.";
     RAY_LOG(DEBUG) << stream.str();
```

Comment:
I know it's not your change, but we should update it to:

```
RAY_LOG(DEBUG) << "Reference for object " << object_id << " has already been freed.";
```

Before:
it'll do string formating no matter what

After:
it'll only have extra overhead when log==DEBUG

Reply:
Hmm this is not the scope of this PR, but sure I will change that.

Reply:
Thanks!
It's always good to just update the easy to improve code like this one along with the PR to improve the overall code quality. Otherwise, I think this probably won't be updated forever 😀